naps/cores/video/gamma_corrector_test.py

```diff
@@ -0,0 +1,51 @@
+import unittest
+from os.path import join, dirname
+import imageio
+from nmigen import *
+from nmigen.sim import Passive
+from naps import SimPlatform
+from naps.stream import write_to_stream
+from naps.cores.video import ImageStream, TableGammaCorrector, write_frame_to_stream, read_frame_from_stream
+
+
+class GammaCorrectorTest(unittest.TestCase):
+    def check_output(self, corrector_gen, gamma):
+        platform = SimPlatform()
+        m = Module()
+
+        input = ImageStream(8)
+        transformer = m.submodules.transformer = corrector_gen(input, gamma)
+        image = imageio.imread(join(dirname(__file__), "wavelet", "che_32.png"))
+
+        # correct the image ourselves to check the corrector's work
+        if gamma != 1:
+            bpp = 8
+            max_pix = 2**bpp - 1
+            lut = list(int(max_pix*((v/max_pix)**gamma)) for v in range(max_pix+1))
+            image_corrected = list(list(lut[pixel] for pixel in line) for line in image)
+        else: # gamma = 1 should not change image at all
+            image_corrected = list(list(pixel for pixel in line) for line in image)
+
+        def write_process():
+            yield from write_frame_to_stream(input, image, pause=False)
+            yield Passive()
+            while True:
+                yield from write_to_stream(input, line_last=0, frame_last=0, payload=0)
+
+        def read_process():
+            result = yield from read_frame_from_stream(transformer.output, timeout=1000, pause=False)
+            imageio.imsave(platform.output_filename_base + "_result.png", result)
```

Comment:
It would be nice, if each of the test cases would not overwrite their outputs.

Reply:
I am not sure what you mean? `platform.output_filename_base` encodes the test name. For example, one image I get from running `python -m unittest gamma_corrector_test.py` is named `.sim_results/GammaCorrectorTest__test_output_table_gamma_corrector_decode__check_output_result.png`. The different test images do not overwrite each other.

Reply:
this is correct

Reply:
Yep, I am blind, sorry.